File: bantu-todos-api/src/test/kotlin/dev/bantu/todos/api/ContractTest.java
package dev.bantu.todos.api;

import com.openpojo.reflection.filters.FilterPackageInfo;
import com.openpojo.validation.Validator;
import com.openpojo.validation.ValidatorBuilder;
import com.openpojo.validation.test.impl.GetterTester;
import com.openpojo.validation.test.impl.SetterTester;
import org.junit.Ignore;
import org.junit.jupiter.api.Test;

public class ContractTest {

    @Test
    @Ignore
    public void testPojo() {
        Validator validator = ValidatorBuilder.create()
            .with(new SetterTester())
            .with(new GetterTester())
            .build();
        validator.validate("dev.bantu.todos.api.model", new FilterPackageInfo());

    }

}

File: bantu-todos-api/src/test/kotlin/dev/bantu/todos/api/ContractTest.kt
package dev.bantu.todos.api

import com.openpojo.validation.ValidatorBuilder
import com.openpojo.validation.test.impl.SetterTester
import com.openpojo.validation.test.impl.GetterTester
import com.openpojo.reflection.filters.FilterPackageInfo
import org.junit.jupiter.api.Test

class ContractTest {

    @Test
    fun testPojo() {
        val validator = ValidatorBuilder.create()
            .with(SetterTester())
            .with(GetterTester())
            .build()
        validator.validate("dev.bantu.todos.api.model", FilterPackageInfo())
    }
}
